src/scenes/Alerts.jsx

```diff
@@ -0,0 +1,241 @@
+import Footer from '../components/Footer';
+import AlertTable from '../components/AlertTable';
+
+import Slider from 'rc-slider';
+import 'rc-slider/assets/index.css';
+
+import api from '../services/api';
+
+import t from '../locale/he_IL';
+import logo from '../assets/logo.png';
+import './Alerts.css';
+import "../../node_modules/leaflet/dist/leaflet.css";
+
+class Alerts extends Component {
+    state = {
+        error: false,
+        loading: false,
+        alerts: [],
+        form: {
+            radius: 3,
+            address: ''
+        },
+
+        // it does not use it, no map when there is no alerts
+        bounds:[{
+            lat:35,
+            lng:35
+        },{
+            lat:25,
+            lng:25
+        }],
+        slider:{
+            min:1,
+            max:10
+        }
+    }
+
+    constructor(props) {
+        super(props);
+        this.handleAddress = this
+            .handleAddress
+            .bind(this);
+        this.handleSlide = this
+            .handleSlide
+            .bind(this);
+        this.handleSubmit = this
+            .handleSubmit
+            .bind(this);
+        this.handleDelete = this
+            .handleDelete
+            .bind(this);
+    }
+    handleSlide(value) {
+        const {form} = this.state;
+        form.radius = value;
+        this.setState({form, error:false});
+    }
+
+    handleAddress(e) {
+        const {form} = this.state;
+        form.address = e.target.value;
+
+        this.setState({form, error:false});
+    }
+
+    handleSubmit(e){
+        const {address, radius} = this.state.form;
+        e.preventDefault();
+        
+        this.setState({loading:true});
+        
+        api.post('/alert',{
+            address, 
+            radius: this.state.slider.max + this.state.slider.min- radius
+        })
+        .then(() => this.getAlerts())
+        .finally(()=>{
+            this.setState({
+                loading: false,
+                 form:{
+                    radius: 3,
+                    address: ''
+            }});
+        })
+        .catch(error => {
+            this.setState({error})
+            console.error(error);
+        })
+    }
+
+    handleDelete(alertId) {
+        api.delete('/alert/'+alertId)
+        .then(() => this.getAlerts())
+    }
+
+    componentDidMount() {
+        this.getAlerts();
+        document.getElementById('homeAddress').focus();
+    }
+
+    getAlerts() {
+        return api
+        .get('/alert')
+        .then(result => {
+            let transparentLayer = leaflet.geoJSON();
+            let fixedBounds = this.state.bounds;
+            if(result.data.length > 0){
+                result.data.map((alert)=>{
+                    leaflet.geoJSON(alert.geom).addTo(transparentLayer);    
+                  })
+                  const layerBounds = transparentLayer.getBounds();
+                  fixedBounds= [layerBounds._southWest, layerBounds._northEast];
```

Comment:
Just put the array in setStage, we don't have to reassign it.

Reply:
fixed